python/tests/test_table.py

```diff
@@ -436,6 +436,19 @@ class MyTable(LanceModel):
     assert result1["text"].iloc[0] != result2["text"].iloc[0]
 
 
+def test_create_scalar_index(db):
+    table = LanceTable.create(
+        db,
+        "my_table",
+        data=pa.Table.from_pydict(
+            {"x": ["c", "b", "a", "e", "b"], "y": [1, 2, 3, 4, 5]}
+        ),
+    )
+    table.create_scalar_index("x")
+    results = table.search().where("x = 'c'").to_arrow()
```

Comment:
1. Sure, I'll add one.

2. What I've been doing in lance is running `explain_plan` and verifying that the appropriate operator is in the plan.  I can do that here too (I keep forgetting I can easily get to a lance dataset if I need to, especially in a test).  Otherwise the use of a scalar index should be completely invisible to the user.

I think there is also a list indices method?  I'll look into this.

3. I agree that might be nice to add at some point.  I guess there is a larger question which is what API users will expect for a vector database.  That being said, I don't think there is any real uniformity here.  Pyarrow users would expect `to_arrow` with a bunch of named arguments (i.e. what we have in lance).  Relational db users would probably expect both vector and scalar queries to be a single method accepting an SQL string.  Pandas users are probably split half a dozen different ways (`read_sql`, `read_table`, `read_parquet`).

Reply:
I extended the existing test with a vector index search and a call to `list_indices` to confirm the index was actually created.